triples-generator/src/main/scala/io/renku/triplesgenerator/events/consumers/tsmigrationrequest/migrations/MigrationToNamedGraphs.scala

```diff
@@ -0,0 +1,175 @@
+/*
+ * Copyright 2022 Swiss Data Science Center (SDSC)
+ * A partnership between École Polytechnique Fédérale de Lausanne (EPFL) and
+ * Eidgenössische Technische Hochschule Zürich (ETHZ).
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.renku.triplesgenerator.events.consumers
+package tsmigrationrequest
+package migrations
+
+import DefaultGraphProjectsFinder.ProjectInfo
+import cats.MonadThrow
+import cats.data.EitherT
+import cats.effect.Async
+import cats.syntax.all._
+import eu.timepit.refined.auto._
+import io.circe.Decoder
+import io.circe.literal._
+import io.renku.events.producers.EventSender
+import io.renku.events.{CategoryName, EventRequestContent}
+import io.renku.graph.model.events.EventStatus.TriplesGenerated
+import io.renku.graph.model.{GraphClass, Schemas, projects}
+import io.renku.metrics.MetricsRegistry
+import io.renku.triplesstore.SparqlQuery.Prefixes
+import io.renku.triplesstore._
+import org.typelevel.log4cats.Logger
+import tooling.{RecordsFinder, RecoverableErrorsRecovery}
+
+private class MigrationToNamedGraphs[F[_]: Async](
+    defaultGraphProjectsFinder: DefaultGraphProjectsFinder[F],
+    namedGraphsProjectFinder:   NamedGraphsProjectFinder[F],
+    eventsSender:               EventSender[F],
+    recoveryStrategy:           RecoverableErrorsRecovery = RecoverableErrorsRecovery
+) extends Migration[F] {
+
+  import defaultGraphProjectsFinder._
+  import eventsSender._
+  import fs2._
+  import namedGraphsProjectFinder._
+  import recoveryStrategy._
+
+  override lazy val name: Migration.Name = MigrationToNamedGraphs.name
+
+  override def run(): EitherT[F, ProcessingRecoverableError, Unit] = EitherT {
+    Stream
+      .iterate(1)(_ + 1)
+      .evalMap(findDefaultGraphProjects)
+      .takeThrough(_.nonEmpty)
+      .flatMap(in => Stream.emits(in))
+      .evalFilterNot { case (projectId, _) => checkExists(projectId) }
+      .map(toRedoEvent)
+      .evalMap { case (payload, ctx) => sendEvent(payload, ctx) }
```

Comment:
I think one could also write `.evalMap(sendEvent.tupled)` to spare the redundant param passing - but just nitpicking here, not important :)

Reply:
It looks like the problem is that the `sendEvent` def is overloaded. When I even try to turn the def into a function with `_` the compiler gets confused.